Makefile

```diff
@@ -85,98 +87,64 @@ test: pre-test
-	$(MAKE) cluster-prepare-wait
-	$(MAKE) deploy
-	$(MAKE) configure-vault
+	$(MAKE) setup-cluster
 	$(MAKE) -C manager run-notebook-readflow-tests
 
 .PHONY: run-notebook-readflow-tls-tests
-run-notebook-readflow-tls-tests: export DOCKER_HOSTNAME?=localhost:5000
-run-notebook-readflow-tls-tests: export DOCKER_NAMESPACE?=fybrik-system
 run-notebook-readflow-tls-tests: export VALUES_FILE=charts/fybrik/notebook-test-readflow.tls.values.yaml
+run-notebook-readflow-tls-tests: export DEPLOY_TLS_CERTS=1
 run-notebook-readflow-tls-tests:
-	$(MAKE) kind
-	$(MAKE) cluster-prepare
-	$(MAKE) docker-build docker-push
-	$(MAKE) -C test/services docker-build docker-push
-	$(MAKE) cluster-prepare-wait
-	cd manager/testdata/notebook/read-flow-tls && ./setup-certs.sh
-	$(MAKE) deploy
-	$(MAKE) configure-vault
+	$(MAKE) setup-cluster
 	$(MAKE) -C manager run-notebook-readflow-tests
 
 .PHONY: run-notebook-readflow-bc-tests
-run-notebook-readflow-tests: export DOCKER_HOSTNAME?=localhost:5000
-run-notebook-readflow-tests: export DOCKER_NAMESPACE?=fybrik-system
 run-notebook-readflow-bc-tests: export VALUES_FILE=charts/fybrik/notebook-test-readflow.values.yaml
+run-notebook-readflow-bc-tests: export DEPLOY_FYBRIK_DEV_VERSION=0
 run-notebook-readflow-bc-tests:
-	$(MAKE) kind
-	$(MAKE) cluster-prepare
-	$(MAKE) docker-build docker-push
-	$(MAKE) -C test/services docker-build docker-push
-	$(MAKE) cluster-prepare-wait
-	$(MAKE) deploy_latest_compatible_CRD_version
-	$(MAKE) configure-vault
+	$(MAKE) setup-cluster
 	$(MAKE) -C manager run-notebook-readflow-tests
 
 .PHONY: run-notebook-writeflow-tests
-run-notebook-writeflow-tests: export DOCKER_HOSTNAME?=localhost:5000
-run-notebook-writeflow-tests: export DOCKER_NAMESPACE?=fybrik-system
 run-notebook-writeflow-tests: export VALUES_FILE=charts/fybrik/notebook-test-writeflow.values.yaml
 run-notebook-writeflow-tests:
-	$(MAKE) kind
-	$(MAKE) cluster-prepare
-	$(MAKE) docker-build docker-push
-	$(MAKE) -C test/services docker-build docker-push
-	$(MAKE) cluster-prepare-wait
-	$(MAKE) deploy
-	$(MAKE) configure-vault
+	$(MAKE) setup-cluster
 	$(MAKE) -C manager run-notebook-writeflow-tests
 
 .PHONY: run-namescope-integration-tests
-run-namescope-integration-tests: export DOCKER_HOSTNAME?=localhost:5000
-run-namescope-integration-tests: export DOCKER_NAMESPACE?=fybrik-system
 run-namescope-integration-tests: export HELM_SETTINGS=--set "clusterScoped=false" --set "applicationNamespace=default"
 run-namescope-integration-tests: export VALUES_FILE=charts/fybrik/integration-tests.values.yaml
 run-namescope-integration-tests:
+	$(MAKE) setup-cluster
+	$(MAKE) -C manager run-integration-tests
+
+.PHONY: setup-cluster
+setup-cluster: export DOCKER_HOSTNAME?=localhost:5000
+setup-cluster: export DOCKER_NAMESPACE?=fybrik-system
+setup-cluster:
 	$(MAKE) kind
 	$(MAKE) cluster-prepare
 	$(MAKE) docker-build docker-push
 	$(MAKE) -C test/services docker-build docker-push
 	$(MAKE) cluster-prepare-wait
+ifeq ($(DEPLOY_TLS_CERTS), 1)
```

Comment:
if we run integration tests on kind, why do we need to docker-push, it can be 
`kind load docker-image  $(IMG)`

Reply:
OK will try that.